pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -341,6 +349,11 @@ func ApplyTaskResultsToPipelineResults(
 				continue
 			}
 			variableParts := strings.Split(variable, ".")
+			// if the referenced task is skipped, we should also skip the results replacements
+			if _, ok := skippedTaskNames[variableParts[1]]; ok {
```

Comment:
Not something to be changed in this PR, but I wonder if having constants for the various indexes of parts and a function for the split would make this code more readable? Like, 

```golang
variableParts := splitResultVariable(variable)
skippedTaskNames[variableParts[VARIABLE_RESULT_TASK_NAME_INDEX]]
```

Reply:
That makes sense, I can fix this